### PR TITLE
[Merged by Bors] - feat: port FieldTheory.Tower

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

- [x] depends on: #3717

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
Note https://github.com/leanprover-community/mathlib/commit/c7bce2818663f456335892ddbdd1809f111a5b72

**Comment by eric-wieser** [comment]
There are actually two commits that are out of sync at https://leanprover-community.github.io/mathlib-port-status/file/field_theory/tower; so it's probably best to rebase and force push once mathport has run tonight

File: Mathlib/FieldTheory/Tower.lean [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    Module.rank F K * Module.rank K A = Module.rank F A := by
+  convert rank_mul_rank' F K A <;> rw [lift_id]
+#align rank_mul_rank rank_mul_rank
+
+namespace FiniteDimensional
+
+open IsNoetherian
+
+theorem trans [FiniteDimensional F K] [FiniteDimensional K A] : FiniteDimensional F A :=
+  Module.Finite.trans K A
```
Comment (eric-wieser) [inline Mathlib/FieldTheory/Tower.lean:79]: I think this needs #3717 to work

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3717
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3717~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by eric-wieser** [comment]
Something very weird is going on here: github won't let me review the diff since the mathport output:

https://github.com/leanprover-community/mathlib4/pull/3716/files/fd2b7f5..8190204

**Comment by eric-wieser** [comment]
bors d+

I'm still struggling to get github to show the diff through the UI; in case anyone else has the same problem, it's https://github.com/leanprover-community/mathlib4/pull/3716/files/fd2b7f5e55a4d7aee4907853c2ad972163792797..043ff9086a1155f6c3c01e762ecce403ae08a627

**Comment by Ruben-VandeVelde** [comment]
Thanks!

bors merge

**Pull request closed without merging**